include/fprotect.h

```diff
@@ -55,6 +55,17 @@ int fprotect_area(uint32_t start, size_t length);
 int fprotect_area_no_access(uint32_t start, size_t length);
 #endif
 
+/**
+ * @brief Check whether a block has already been protected.
+ *
+ * @param[in]  addr  The address to check. The block containing this address
+ *                   will be checked.
+ *
+ * @retval true   If protected
```

Comment:
Should this interface be made in such a way that it might be possible to express whether a region is protected against reads, writes and execution?

I'm not sure if its possible to design a solid API around this given that some of the HW might not support reading this level of detail. Anyway, just a suggestion, it might not be useful if the users of this function typically call it to check if there is "any" protection configured.

Reply:
Take a look now.